### PR TITLE
syslog-ng: use syslog-ng-ctl to handle reload operation

### PR DESCRIPTION
**Pull request by pprindeville**
Maintainer: @MikePetullo 
Compile tested: x86_64, generic, LEDE HEAD (6bdb662)
Run tested: same

Built package.  Stopped service on test box.  Installed .ipk on test box.  Restarted service.

Description:

Fix `reload` command in init.d script (by using `syslog-ng-ctl reload` to handle it).

Retired `/etc/custom-logs.conf` in favor of `/etc/syslog-ng.d` directory.

Add parsing of config files to `start` command in init.d script.

### CODE REVIEW

**Comment by pprindeville** [comment]
Had to nix `PKG_MD5SUM` to get Travis to pass.  Sigh.

File: admin/syslog-ng/Makefile [hunk 1]
```diff
@@ -31,7 +30,8 @@ define Package/syslog-ng/description
 endef
 
 define Package/syslog-ng/conffiles
-  /etc/syslog-ng.conf
+/etc/syslog-ng.conf
+/etc/syslog-ng.d/
 endef
```
Comment (stintel) [inline admin/syslog-ng/Makefile]: Seems unrelated to the commit message. Please split it into another commit.

File: admin/syslog-ng/files/syslog-ng.init [hunk 2]
```diff
@@ -4,16 +4,26 @@
 START=20
 
 SERVICE_USE_PID=1
+SERVICE_PID_FILE=/var/run/syslog-ng.pid
+
+config_file=/etc/syslog-ng.conf
```
Comment (stintel) [inline admin/syslog-ng/files/syslog-ng.init]: Also unrelated to the commit message, separate commit please.

File: admin/syslog-ng/Makefile [hunk 3]
```diff
@@ -69,8 +69,9 @@ define Package/syslog-ng/install
 		install-moduleLTLIBRARIES DESTDIR="$(1)"
 	$(INSTALL_DIR) $(1)/etc/init.d
 	$(INSTALL_BIN) ./files/syslog-ng.init $(1)/etc/init.d/syslog-ng
-	$(INSTALL_DIR) $(1)/etc
+	$(INSTALL_DIR) $(1)/etc/syslog-ng.d
 	$(INSTALL_DATA) ./files/syslog-ng.conf $(1)/etc
+	$(INSTALL_DATA) ./files/custom-logs.conf $(1)/etc/syslog-ng.d/
```
Comment (stintel) [inline admin/syslog-ng/Makefile]: Do we really need this file? Having an empty syslog-ng.d directory seems good enough to me.

File: admin/syslog-ng/Makefile (hunk 1 above)
Comment (pprindeville) [inline admin/syslog-ng/Makefile]: Fixed

File: admin/syslog-ng/Makefile (hunk 3 above)
Comment (pprindeville) [inline admin/syslog-ng/Makefile]: Fixed

File: admin/syslog-ng/files/syslog-ng.init (hunk 2 above)
Comment (pprindeville) [inline admin/syslog-ng/files/syslog-ng.init]: Fixed

**Comment by pprindeville** [comment]
Redux per @stintel's review.

**Comment by pprindeville** [comment]
There's some Travis weirdness with the new checks.  Ignore the CI failure.

**Comment by stintel** [comment]
There are still commits that contain changes unrelated to the commit message. Also, I didn't notice that this does not use procd yet. Since you are working on it, it might be better to convert it to procd as well. You can then drop SERVICE_PID_FILE and SERVICE_USE_PID entirely, and use --process-mode foreground.

Splitting this in logical changes, I would probably end up doing it like this:
1. syslog-ng: remove PKG_MD5SUM
2. syslog-ng: install/include /etc/syslog-ng.d
3. syslog-ng: use variables for paths in init script
4. syslog-ng: convert to procd
5. syslog-ng: add syntax check to init script
6. syslog-ng: reload with syslog-ng-ctl

Keep in mind to keep the subject lines <=50 characters.

Example of how I did a similar series for net-snmp: e2b43afc857f9204f1ff234500d533665b8c8d35, 8faff3334f319f8a198e0da31b7f25263e407919, 53fc8d3191cf9dc712cab9d7ae6a4b3d179fbdd1, bee2865492fa8321a74627fd15a24c7caf54790d

**Comment by pprindeville** [comment]
The way that procd interacts with foreground mode syslog-ng creates a race condition. That can't be fixed in this PR.

**Comment by stintel** [comment]
Ok, so replace 4 from previous comment with syslog-ng: get rid of guardian process.

**Comment by pprindeville** [comment]
Redux-bis.

**Comment by stintel** [comment]
Almost perfect, super easy to review, just 1 more thing:
> Keep in mind to keep the subject lines <=50 characters.

If you can't explain it completely in the subject line, keep it short like my examples and explain further in commit message. If you use vim as editor for the commit message, it should show text in red as soon as it's too long. I can highly recommend it.

**Comment by pprindeville** [comment]
> Keep in mind to keep the subject lines <=50 characters.

Fixed

**Comment by pprindeville** [comment]
@MikePetullo Are we good to go when Stijn signs off?  Thanks

**Comment by MikePetullo** [comment]
Looks good.
